src/Discord.Net.WebSocket/DiscordShardedClient.cs

```diff
@@ -519,6 +519,49 @@ private void RegisterEvents(DiscordSocketClient client, bool isPrimary)
             client.WebhooksUpdated += (arg1, arg2) => _webhooksUpdated.InvokeAsync(arg1, arg2);
             client.AuditLogCreated += (arg1, arg2) => _auditLogCreated.InvokeAsync(arg1, arg2);
         }
+
+        public async Task<SocketApplicationCommand> CreateGlobalApplicationCommandAsync(ApplicationCommandProperties properties, RequestOptions options = null)
+        {
+            var model = await InteractionHelper.CreateGlobalCommandAsync(this, properties, options).ConfigureAwait(false);
+
+            SocketApplicationCommand entity = null;
+
+            foreach (var shard in _shards)
+            {
+                entity = shard.State.GetOrAddCommand(model.Id, (id) => SocketApplicationCommand.Create(shard, model));
+            }
+
+            System.Diagnostics.Debug.Assert(entity != null, "There should be at least one shard to get the entity");
+
+            //Update it in case it was cached
+            entity.Update(model);
```

Comment:
what would happen if the command is cached in multiple shard states? it would be updated in only one shard with this impl

Reply:
Updated to all shards. I wonder though if I should return the last entity (like the current implementation).